### ISSUE TITLE
`build`: restore `pkgrel` on --rebuild failure

### ISSUE THREAD

**Issue by AladW** [AladW/aurutils#1050]
If `--rebuild` is specified with a `pkgrel` increment, and any failure happens in `makepkg` (e.g. a GPG key that was not imported), the bumped `pkgrel` is kept. After fixing the error, subsequent runs will have useless `pkgrel` increases.

Additionally, when a `pkgver()` function modifies `pkgver`, `pkgrel` is reset to 1. This should likely be documented.

**Comment by AladW** [comment]
Another issue with `--rebuild` is that it's not thread-safe, since it writes to the PKGBUILD in a shared source directory. `pkgver()` functions have the same problem. It hardly makes sense to rebuild the same package in parallel, but this is another thing that should be documented.
